File: basic/10-two-sum/10-two-sum.java
import java.util.HashMap;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[]{numMap.get(complement), i};
            }
            numMap.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] result = twoSum(nums, target);
        if (result[0] != -1) {
            System.out.printf("Indices: %d, %d%n", result[0], result[1]);
        } else {
            System.out.println("No solution found.");
        }
    }
}

File: basic/10-two-sum/10-two-sum.c
#include <stdio.h>

void two_sum(int* nums, int numsSize, int target) {
  for (int i = 0; i < numsSize; i++) {
    for (int j = i + 1; j < numsSize; j++) {
      if (nums[i] + nums[j] == target) {
        printf("%d, %d", i, j);
        return;
      }
    }
  }
  print("No result");
}

int main() {
  int nums[] = {1, 2, 8, 10};
  int target = 9;
  int size = sizeof(nums) / sizeof(nums[0]);
  two_sum(nums, size, target);
  return 0;
}
